framework/include/variables/MooseVariableData.h

```diff
@@ -725,3 +723,25 @@ MooseVariableData<OutputType>::adUDotDot() const
 
   return _ad_u_dotdot;
 }
+
+template <typename OutputType>
+const ADTemplateVariablePhiGradient<typename MooseVariableData<OutputType>::OutputShape> &
+MooseVariableData<OutputType>::adGradPhi() const
+{
+  if (_element_type == Moose::ElementType::Neighbor ||
+      _element_type == Moose::ElementType::Neighbor)
```

Comment:
What's this if condition checking for? Same question for adGradPhiFace.

Reply:
oopsie. Typo